### ISSUE TITLE
Add ability to stream or push items into a Quickpick

### ISSUE THREAD

**Issue by eamodio** [Microsoft/vscode#21372]
- VSCode Version: `1.10.0-insider` 256d76fed7947e7289f921df10658d7a69233b65
- OS Version: macOS Sierra `10.12.3 (16D32)`

Basically provide a way to add additional items to a quickpick after it is shown and/or provide a way to pass an array of promises to `showQuickPick` that would add items as each one resolves.

**Comment by jrieken** [comment]
> way to pass an array of promises to showQuickPick that

We do allow a single promise that resolves to an array of strings or message item, like so 

```ts
const myPromisedItems = new Promise(resolve => resolve(['one', 'two', 'three']);
vscode.window.showQuickPick(myPromisedItems)
```

What we don't support is to update the quick pick later, neither with a promise nor with a sync set of items

**Comment by eamodio** [comment]
Yeah I was thinking with the array of promises vs a single, is that there could be a race setup where anytime one of the promises was resolved it would add its items to the list -- so entries would somewhat stream in.

For example in my [Find Related](https://github.com/eamodio/vscode-find-related) extension, I kick of a series of promises to search for different file matches, and now currently aggregate them into a single promise that generates the list of quickpick items, but it would be great if I could have the items returned by each promise added to the list when it became ready. So longer searches don't block the shorter ones.

**Comment by jrieken** [comment]
Understood. The big challenge with streaming results (to basically anywhere) is how the UI should behave. Once a subset of results is in users can filter and select one and it will be hard to keep the UX stable when more results come in, esp wrt sorting

**Comment by eamodio** [comment]
Yeah, that is tricky to be sure.

**Comment by KnisterPeter** [comment]
Just as a reference to track the issue KnisterPeter/vscode-github#84
